pkg/auth/ldaputil/query.go

```diff
@@ -131,16 +81,28 @@ func NewLDAPQueryOnAttribute(config api.LDAPQuery, attribute string) (LDAPQueryO
 	}, nil
 }
 
+// LDAPQueryOnAttribute encodes an LDAP query that conjoins two filters to extract a specific LDAP entry
+// This query is not self-sufficient and needs the value of the QueryAttribute to construct the final filter
+type LDAPQueryOnAttribute struct {
+	// Query retrieves entries from an LDAP server
+	LDAPQuery
+
+	// QueryAttribute is the attribute for a specific filter that, when conjoined with the common filter,
+	// retrieves the specific LDAP entry from the LDAP server. (e.g. "cn", when formatted with "aGroupName"
+	// and conjoined with "objectClass=groupOfNames", becomes (&(objectClass=groupOfNames)(cn=aGroupName))")
+	QueryAttribute string
+}
+
 // NewSearchRequest creates a new search request from the identifying query by internalizing the value of
 // the attribute to be filtered as well as any attributes that need to be recovered
 func (o *LDAPQueryOnAttribute) NewSearchRequest(attributeValue string, attributes []string) (*ldap.SearchRequest, error) {
 	if strings.EqualFold(o.QueryAttribute, "dn") {
-		if !strings.Contains(attributeValue, o.BaseDN) {
-			return nil, &errQueryOutOfBounds{QueryDN: attributeValue, BaseDN: o.BaseDN}
-		}
 		if _, err := ldap.ParseDN(attributeValue); err != nil {
 			return nil, fmt.Errorf("could not search by dn, invalid dn value: %v", err)
 		}
+		if !strings.Contains(attributeValue, o.BaseDN) {
+			return nil, NewQueryOutOfBoundsError(attributeValue, o.BaseDN)
+		}
```

Comment:
Justification for re-order here:
If something's happened and we created a malformed DN, the `Contains` check will almost always fail. I think it will be more useful to know that you are about to use a DN that isn't viable first, some of the errors I was able to generate were useless in that they told me my issue was out of bounds, but in reality it was malformed DN.